### PR TITLE
Fix/sprint3 main page

### PR DESCRIPTION
**Pull request by RayMet**

### CODE REVIEW

File: client/src/components/HomePage/Registration.jsx [hunk 1]
```diff
@@ -15,10 +15,41 @@ class Registration extends Component {
...
+    onChangeHandlerPassword = (e) => {
     let { id } = e.currentTarget;
-    this.setState({ [id]: e.currentTarget.value });
-  };
+    this.setState({ [id]: e.currentTarget.value },
+    () => {
+      if (this.state.password === this.state.repeat_password) this.setState({ check_password: "" });
+      else {
+        this.setState({ check_password: "passwords isn't equal" });
+        return null;
```
Comment (Merteg) [inline client/src/components/HomePage/Registration.jsx]: It's a useless line, this function will already end here without it.

File: client/src/components/HomePage/Registration.jsx [hunk 2]
```diff
@@ -15,10 +15,41 @@ class Registration extends Component {
...
+      }
+    });
+    }
+
+    onChangeHandlerEmail = (e) => {
+    let { id } = e.currentTarget;
+    this.setState({ [id]: e.currentTarget.value },
+    () => {
+      if (!emailValidation(this.state.email)) {
+        this.setState({ check_email: "Please provide a valid email address" });
```
Comment (Merteg) [inline client/src/components/HomePage/Registration.jsx]: I know you have a problem with the length of the error message, you can just shorten it to something like "Invalid email"

File: client/src/components/HomePage/Registration.jsx [hunk 3]
```diff
@@ -29,13 +60,17 @@ class Registration extends Component {
...
+      window.Materialize.toast("Invalid input", 3000);
+    }
+    if (!emailValidation(this.state.email)) {
+      this.setState({ check_email: "Please provide a valid email address" });
+    }
+    if (!this.state.nickname.length) {
+      this.setState({ check_nickname: "Empty field" });
       window.Materialize.toast("Invalid input", 3000);
-      return null;
     }
```
Comment (Merteg) [inline client/src/components/HomePage/Registration.jsx]: I think better to stop function if you have some errors before sending a request to the server,

File: client/src/components/HomePage/Registration.jsx [hunk 4]
```diff
@@ -45,10 +80,12 @@ class Registration extends Component {
 
         if ('error' in data) {
           this.setState({ [`check_${data.error}`]: `Invalid ${data.error}` })
-          this.props.history.push('/registration');
           return null;
+
         } else {
-          this.props.history.push('/');
+        window.Materialize.toast("Check your email box", 3000);
+        window.history.forward('/registration-info');
```
Comment (Merteg) [inline client/src/components/HomePage/Registration.jsx]: Wrong indentation. Also, you don't need this `return null`, you don't have any code after this `if`.

File: client/src/components/HomePage/Registration.jsx [hunk 5]
```diff
@@ -57,13 +94,13 @@ class Registration extends Component {
 
   render() {
     return (
-      <div className="white">
+      <div className="white" id="registration-container">
```
Comment (Merteg) [inline client/src/components/HomePage/Registration.jsx]: Why did you add this id?

File: client/src/components/HomePage/Registration.jsx (hunk 4 above)
Comment (Merteg) [inline client/src/components/HomePage/Registration.jsx]: I don't think it works in right way, better to use `this.props.history.push`

File: client/src/components/RegistrationInfo.jsx [hunk 6]
```diff
@@ -0,0 +1,20 @@
+import React, { Component } from 'react';
+import {Col, Row} from "react-materialize";
+
+class RegistrationInfo extends Component{
```
Comment (Merteg) [inline client/src/components/RegistrationInfo.jsx]: If your component just renders some markup, better to use a function declaration. [Style Guide](https://github.com/airbnb/javascript/tree/master/react#class-vs-reactcreateclass-vs-stateless)

File: server/server/models/user.py [hunk 7]
```diff
@@ -50,6 +50,20 @@ def get_one(cls, request, **kwargs):
...
 
+    @classmethod
+    def get_user_by_email(cls, request, email):
+        """ """
+        return request.dbsession.query(User)\
+            .filter(func.lower(User.email) == func.lower(email)).one_or_none()
+
+    @classmethod
+    def get_user_by_nickname(cls, request, nickname):
+        """ """
```
Comment (Merteg) [inline server/server/models/user.py]: Add a docstrings

File: server/server/views/auth.py [hunk 8]
```diff
@@ -25,7 +26,9 @@ def log_in_post(request):
         'success': False,
         'user': {}
     }
-    user = User.get_one(request, email=request.validated['email'])
+    user = request.dbsession.query(User)\
+        .filter(func.lower(User.email) == func.lower(request.validated['email']))\
+        .one_or_none()
```
Comment (Merteg) [inline server/server/views/auth.py]: This does not meet the PEP8 standard. (W503, E128)

**Review by Merteg — CHANGES_REQUESTED** [review]

**Comment by Merteg** [comment]
Also, you have merge conflicts

File: client/src/components/HomePage/Registration.jsx (hunk 1 above)
Comment (RayMet) [reply client/src/components/HomePage/Registration.jsx]: What is line useless? 22?

File: client/src/components/HomePage/Registration.jsx (hunk 5 above)
Comment (RayMet) [reply client/src/components/HomePage/Registration.jsx]: Why not? All elements must have their id especially div. But main reason was i tried some css with errors

File: client/src/components/HomePage/Registration.jsx (hunk 4 above)
Comment (RayMet) [reply client/src/components/HomePage/Registration.jsx]: This.props.history.push do not working. And  window.history.forward works fine, I read a docks and made this lines.

File: client/src/components/RegistrationInfo.jsx (hunk 6 above)
Comment (RayMet) [reply client/src/components/RegistrationInfo.jsx]: I dont see here problems. Like Serhiy said that must be separate state page.

File: client/src/components/HomePage/Registration.jsx (hunk 1 above)
Comment (Merteg) [reply client/src/components/HomePage/Registration.jsx]: 25

File: client/src/components/HomePage/Registration.jsx (hunk 5 above)
Comment (Merteg) [reply client/src/components/HomePage/Registration.jsx]: Now you don't have css for this. `All elements must have their id`, who told you this?

File: client/src/components/HomePage/Registration.jsx (hunk 4 above)
Comment (Merteg) [reply client/src/components/HomePage/Registration.jsx]: Okay, maybe it's don't work here because of Route in Route, so I don't have any questions for this

File: client/src/components/RegistrationInfo.jsx (hunk 6 above)
Comment (Merteg) [reply client/src/components/RegistrationInfo.jsx]: You don't understand me, I told about function declaration of a component. Please follow the link

File: client/src/components/HomePage/Registration.jsx (hunk 5 above)
Comment (RayMet) [reply client/src/components/HomePage/Registration.jsx]: I can delete this. But I dont see a problem for give id. Or just proof that are wrong.

Comment (Merteg) [reply client/src/components/HomePage/Registration.jsx]: The first [link](https://css-tricks.com/the-difference-between-id-and-class/#article-header-id-9) in google. You can find more, but "If you don't need them, don't use them".
The [article](https://dev.to/claireparker/reasons-not-to-use-ids-in-css-4ni4) why `class` is better than `id` in css. If you need css - use class, if you need js - use id.

**Comment by RayMet** [comment]
Update

File: client/src/components/HomePage/Registration.jsx (hunk 3 above)
Comment (Merteg) [reply client/src/components/HomePage/Registration.jsx]: This can be optimized. You can create bool variable and check it if you have an errors instead of doing validation two times.

File: client/src/components/HomePage/Registration.jsx [hunk 9]
```diff
@@ -81,20 +83,17 @@ class Registration extends Component {
         if ('error' in data) {
           this.setState({ [`check_${data.error}`]: `Invalid ${data.error}` })
           return null;
```
Comment (Merteg) [inline client/src/components/HomePage/Registration.jsx]: I told about all `if` statement, not also about `else`.

File: server/server/models/user.py [hunk 10]
```diff
@@ -52,13 +52,17 @@ def get_one(cls, request, **kwargs):
...
+        """
         return request.dbsession.query(User)\
             .filter(func.lower(User.email) == func.lower(email)).one_or_none()
 
     @classmethod
     def get_user_by_nickname(cls, request, nickname):
-        """ """
+        """Get one user from db by nickname
+        Return user object if user exist and return None if not
+        """
```
Comment (Merteg) [inline server/server/models/user.py]: It does not up to the PEP257 standard. [PEP257](https://www.python.org/dev/peps/pep-0257/#multi-line-docstrings)

**Comment by RayMet** [comment]
Update, please review.

File: server/server/models/user.py (hunk 10 above)
Comment (Merteg) [reply server/server/models/user.py]: > Multi-line docstrings consist of a summary line just like a one-line docstring, followed by a blank line, followed by a more elaborate description.

**Comment by RayMet** [comment]
Update

**Review by Merteg — APPROVED** [review]

**Review by yazhyvyi — APPROVED** [review]
Nice
